### PR TITLE
Port connector-specific field validation to the new system

### PR DESCRIPTION
**Pull request by tomhoule**
This is one of the two validations still holding the validation pipeline
simplification back.

### CODE REVIEW

File: libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs [hunk 1]
```diff
@@ -45,18 +45,6 @@ impl Connector for MongoDbDatamodelConnector {
...
-        // If the field is _not_ a native-type-annotated field and it has a `dbgenerated` default, we error.
-        if !matches!(field.field_type(), FieldType::Scalar(_, _, Some(_)))
-            && matches!(field.default_value().as_ref().map(|v| v.kind()), Some(DefaultKind::Expression(expr)) if expr.is_dbgenerated())
-        {
-            errors.push(ConnectorError::from_kind(ErrorKind::FieldValidationError {
-                field: field.name().to_owned(),
-                message: "MongoDB `@default(dbgenerated())` fields must have a native type annotation.".to_owned(),
-            }));
-        }
-    }
```
Comment (tomhoule) [inline libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs:58]: This is potentially controversial. This is not tested, and the purpose isn't completely clear. We can of course reimplement if it turns out we _do_ wan to validate this.

**Comment by tomhoule** [comment]
This will shortly be followed by something similar for the last validation. Massive simplifications possible after that.

File: libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs (hunk 1 above)
Comment (tomhoule) [reply libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs:58]: I think we'd want a connector method for defaults validation, but I can't reimplement a validation I don't know and that isn't tested.

Comment (tomhoule) [reply libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs:58]: Ideally we'd validate exactly this. As it is, the migration will probably break at some point and it's not clear why we always forbid that. Btw we were trying to get rid of `dbgenerated()` without argument, and now we're relying on it by design on mongo. I'm not a fan at all.

Comment (tomhoule) [reply libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs:58]: Test and new code in a68199c

**Review by pimeys — APPROVED** [review]

File: libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs (hunk 1 above)
Comment (pimeys) [reply libs/datamodel/connectors/mongodb-datamodel-connector/src/lib.rs:58]: Yeah I don't like this either. I also don't like to not have a validation for this. We should aim for better design with the dbgenerated for sure.

**Pull request merged**